### PR TITLE
Feat:TEA-226/converted uuids of Zuri Chat to shorter slugs

### PR DESCRIPTION
**Pull request by MrTaaJ**
# Fixes Issue/Linear Ticket

**My PR Fixes TEA-226/converted uuids of Zuri Chat to shorter slugs**

# Changes proposed

## What were you told to do?
Description: Zuri Task The urls of ZuriChat are too long. There are uuids in there. We need to switch from uuid to shorter slugs. e.g zuri.chat/marks-space-v5Wd/random

## What did you do?
converted uuids of Zuri Chat to shorter slugs

# Check List (Check all the applicable boxes)

🚨Please review the [style guide for contributing](https://github.com/zurichat/zc_main/blob/dev/docs/STYLING.md) and [guidelines for contributing](https://github.com/zurichat/zc_main/blob/dev/docs/CONTRIBUTING.md) to this repository.

- [X] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.
- [X] I am making a pull request against the **dev branch** (left side).
- [X] My commit messages styles matches our requested structure.
- [X] My code additions will fail neither code linting checks nor unit test.
- [X] I am only making changes to files I was requested to.

# Screenshots/ Videos
zuri-chat-1
zuri-chat-2
zuri-chat-3
zuri-chat-4
zuri-chat-5

### CODE REVIEW

File: packages/main/src-old/pages/createworkspace/components/CompanyName.js [hunk 1]
```diff
@@ -64,7 +64,7 @@ function CompanyName({ input }) {
             });
         }, 500);
 
-        localStorage.setItem("currentWorkspace", newOrgId);
+        // localStorage.setItem("currentWorkspace", newOrgId);
```
Comment (JulianaSau) [inline packages/main/src-old/pages/createworkspace/components/CompanyName.js:67]: Remove commented out code

File: packages/main/src-old/pages/admin/AboutWorkSpace.js [hunk 2]
```diff
@@ -8,7 +8,9 @@ import { authAxios } from "./Utils/api";
 const AboutWorkSpace = () => {
   const [orgDetails, setOrgdetails] = useState({});
   const [items, setItems] = useState(null);
-  const currentWorkspace = localStorage.getItem("currentWorkspace");
+  // const currentWorkspace = localStorage.getItem("currentWorkspace");
+
+  const currentWorkspace = localStorage.getItem("currentWorkspace") || null;
```
Comment (JulianaSau) [inline packages/main/src-old/pages/admin/AboutWorkSpace.js:13]: Now what happens when the current workspace is `null`?
And why are you making an edit on a file in `src-old`. We're working with files in the `src` folder only

File: packages/main/src/components/protected/sidebar/components/PluginRoomAddUser.jsx [hunk 3]
```diff
@@ -15,7 +15,9 @@ const PluginRoomAddUser = ({ isOpen, isClosed, room_id }) => {
   // const channelName =url.split("https://zuri.chat/").pop())
   const channelName = url.split("http://localhost:9000/").pop();
   const baseUrl = "zuri.chat";
-  const workspace = sessionStorage.getItem("currentWorkspace");
+
+  // const workspace = sessionStorage.getItem("currentWorkspace");
```
Comment (JulianaSau) [inline packages/main/src/components/protected/sidebar/components/PluginRoomAddUser.jsx:19]: Remove all instances of commented out code if theyre not being used

File: packages/ui/src/message-room-details-dialog/MessageRoomDetailsDialog.jsx [hunk 4]
```diff
@@ -298,7 +298,7 @@ export function MembersPanel({ config }) {
 
     const theUserData = JSON.parse(localStorage.getItem("userData"));
 
-    console.log(theUserData.user.org_id);
+    console.log(theUserData.user.org_id, "This is where I don't know");
```
Comment (JulianaSau) [inline packages/ui/src/message-room-details-dialog/MessageRoomDetailsDialog.jsx:301]: why the console.log? remove it

File: packages/root-config/src/zuri-root-config.js [hunk 5]
```diff
@@ -50,10 +53,14 @@ window.addEventListener("zuri-plugin-load", () => {
...
-                "currentWorkspace"
-              )}/${appName.replace("zuri-", "")}`
+              baseName: `/workspace/${currentWorkspace}/${appName.replace(
+                "zuri-",
+                ""
+              )}`
             }
+            // baseName: `/workspace/${localStorage.getItem(
+            //   "currentWorkspace"
+            // )}/${appName.replace("zuri-", "")}`
```
Comment (JulianaSau) [inline packages/root-config/src/zuri-root-config.js:63]: Remove commented out code

File: packages/root-config/src/zuri-root-config.js [hunk 6]
```diff
@@ -33,9 +35,10 @@ window.addEventListener("zuri-plugin-load", () => {
           activeWhen: pathToActiveWhen(`/workspace/:id/${plugin.pluginPath}`),
           customProps: {
             domElement: document.getElementById("zuri-plugin-load-section"),
-            baseName: `/workspace/${localStorage.getItem("currentWorkspace")}/${
-              plugin.pluginPath
-            }`
+
+            baseName: `/workspace/${currentWorkspace}/${plugin.pluginPath}`
+            // baseName: `/workspace/${localStorage.getItem("currentWorkspace")}/${
+            //   plugin.pluginPath
```
Comment (JulianaSau) [inline packages/root-config/src/zuri-root-config.js:41]: Remove commented out code

**Review by JulianaSau — CHANGES_REQUESTED** [review]
Remove all instances of commented-out codes, console logs and look at the changes requested

File: packages/main/src-old/pages/admin/AboutWorkSpace.js (hunk 2 above)
Comment (MrTaaJ) [reply packages/main/src-old/pages/admin/AboutWorkSpace.js:13]: I was mistaken, I made an edit on it before, so I had to leave it as I met it.

**Comment by MrTaaJ** [comment]
The changes have been implemented. Thank you.

File: packages/main/src-old/pages/createworkspace/components/WorkspaceHome.js [hunk 7]
```diff
@@ -43,6 +43,8 @@ const WorkspaceHome = () => {
 
           //Store user copy in localstorage
           sessionStorage.setItem("user", JSON.stringify(data.user));
+
+          console.log(data.user, "I am consoling data from WorkspaceHome.jsx");
```
Comment (JulianaSau) [inline packages/main/src-old/pages/createworkspace/components/WorkspaceHome.js:47]: Remove this console.log

File: packages/main/src/components/protected/sidebar/components/RoomItem.jsx [hunk 8]
```diff
@@ -10,15 +10,21 @@ import { useRouteMatch } from "react-router-dom";
 import { useTranslation } from "react-i18next";
 
 const RoomItem = ({ room, baseUrl, pluginId }) => {
+  // let currentWorkspace = localStorage.getItem("currentWorkspace");
```
Comment (JulianaSau) [inline packages/main/src/components/protected/sidebar/components/RoomItem.jsx:13]: Commented out code

File: packages/main/src/components/protected/sidebar/utils/get-user-info.js [hunk 9]
```diff
@@ -1,7 +1,10 @@
 import { authAxios } from "./Api";
 export const GetUserInfo = async () => {
   let user = JSON.parse(sessionStorage.getItem("user"));
-  const currentWorkspace = localStorage.getItem("currentWorkspace");
+
+  const currentWorkspace = localStorage.getItem("currentWorkspace") || null;
+
+  // const currentWorkspace = localStorage.getItem("currentWorkspace");
```
Comment (JulianaSau) [inline packages/main/src/components/protected/sidebar/utils/get-user-info.js:7]: Remove commented out code

File: packages/main/src/components/protected/topbar/components/Profile.jsx [hunk 10]
```diff
@@ -42,7 +42,7 @@ const Profile = () => {
   useEffect(() => {
     if (currentWorkspace) {
       authAxios
-        .get(`/organizations/${currentWorkspace}`)
+        .get(`v`)
```
Comment (JulianaSau) [inline packages/main/src/components/protected/topbar/components/Profile.jsx:45]: Where is this `v` coming from?

File: packages/main/src/components/protected/topbar/components/TopbarModal.jsx [hunk 11]
```diff
@@ -100,6 +100,8 @@ const TopbarModal = ({ members, statusModal, setStatusModal }) => {
     setPresence
   } = state;
 
+  // const workspace = sessionStorage.getItem("currentWorkspace");
```
Comment (JulianaSau) [inline packages/main/src/components/protected/topbar/components/TopbarModal.jsx:103]: Remove this commented out code

File: packages/main/src/components/user-organisations-listing/UserOrganisationsListing.jsx [hunk 12]
```diff
@@ -4,12 +4,54 @@ import { useTranslation } from "react-i18next";
...
+    setNewOrganizations([...hhh]);
+  }, [organizations]);
+
+  // const urlIds
+  // console.log(JSON.parse(urlIds).workspaceIds)
+  // const newOrganizations = [...organizations]
+
+  // newOrganizations.map(org => {
+  //   org['short_id'] = urlIds.filter(urlId => urlId === org.id)[0].short_id
+  // })
```
Comment (JulianaSau) [inline packages/main/src/components/user-organisations-listing/UserOrganisationsListing.jsx:52]: Remove all that commented out code

File: packages/main/src/pages/protected/create-workspace/LaunchWorkspace.jsx [hunk 13]
```diff
@@ -30,6 +30,7 @@ export default function Index({ createWorkspaceData }) {
     );
     console.log(createWorkspaceApiCall);
     const workspaceId = createWorkspaceApiCall.data.data.organization_id;
+    // const workspaceId = "tr456vdh";
```
Comment (JulianaSau) [inline packages/main/src/pages/protected/create-workspace/LaunchWorkspace.jsx:33]: Remove this

File: packages/main/src/pages/protected/create-workspace/LaunchWorkspace.jsx [hunk 14]
```diff
@@ -70,7 +71,20 @@ export default function Index({ createWorkspaceData }) {
...
+
+    const short_id = `${workspaceId.slice(4, 6)}${workspaceId.slice(
+      6,
+      8
+    )}${workspaceId.slice(-3, -1)}`;
+
     localStorage.setItem("currentWorkspace", workspaceId);
+    localStorage.setItem("currentWorkspaceShort", short_id);
+
+    console.log(workspaceId, short_id, "Checking for IDS");
```
Comment (JulianaSau) [inline packages/main/src/pages/protected/create-workspace/LaunchWorkspace.jsx:85]: Remove the console

File: packages/main/src/pages/protected/create-workspace/LaunchWorkspace.jsx [hunk 15]
```diff
@@ -70,7 +71,20 @@ export default function Index({ createWorkspaceData }) {
...
+
+    // localStorage.setItem("currentWorkspace", workspaceId);
```
Comment (JulianaSau) [inline packages/main/src/pages/protected/create-workspace/LaunchWorkspace.jsx:87]: Remove commented out code

File: packages/main/src/pages/protected/create-workspace/Step0.jsx [hunk 16]
```diff
@@ -22,6 +22,7 @@ export default function Index() {
       { headers: { Authorization: `Bearer ${user.token}` } }
     );
     const { data } = result.data;
+    // console.log(data, 'from step 0 .jsxxxxxxxxxxxxxxxxxxxxx')
```
Comment (JulianaSau) [inline packages/main/src/pages/protected/create-workspace/Step0.jsx:25]: Remove this

File: packages/main/src/pages/protected/workspace/useParamHook.jsx [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+            [key]: urlIdes[`${key}s`]?.filter(
+              urlId => urlId.short_id === params[key]
+            )[0]?.real_id,
+            short_id: params[key]
+          };
+        }
+
+        // if (!urlIdes) {
+        //   urlIds[key] = params[key]
+        // }
```
Comment (JulianaSau) [inline packages/main/src/pages/protected/workspace/useParamHook.jsx:35]: Remove commented out code

**Review by JulianaSau — CHANGES_REQUESTED** [review]
My question still stands, you set current workspace to null if the current workspace ID isn't set in the localStorage. Will tge code crash if the current workspace ID is null

File: packages/main/src-old/pages/createworkspace/components/WorkspaceHome.js (hunk 7 above)
Comment (MrTaaJ) [reply packages/main/src-old/pages/createworkspace/components/WorkspaceHome.js:47]: Done

File: packages/main/src/components/protected/sidebar/utils/get-user-info.js (hunk 9 above)
Comment (MrTaaJ) [reply packages/main/src/components/protected/sidebar/utils/get-user-info.js:7]: Done

File: packages/main/src/components/protected/topbar/components/Profile.jsx (hunk 10 above)
Comment (MrTaaJ) [reply packages/main/src/components/protected/topbar/components/Profile.jsx:45]: I seriously didn't touch any Axios call. I don't know how it happens to be my changes, probably when I was merging. I was only working on the data that should be in currentWorkspace and when to use the long URL for axios. But the way it is, I don't think it is needed, so I will remove it.

Comment (MrTaaJ) [reply packages/main/src/components/protected/topbar/components/Profile.jsx:45]: I just reviewed the code very well, I think probably my keyboard or something, but it wasn't intentional. So, I have reverted back to this

File: packages/main/src/components/protected/topbar/components/TopbarModal.jsx (hunk 11 above)
Comment (MrTaaJ) [reply packages/main/src/components/protected/topbar/components/TopbarModal.jsx:103]: Done

File: packages/main/src/components/user-organisations-listing/UserOrganisationsListing.jsx (hunk 12 above)
Comment (MrTaaJ) [reply packages/main/src/components/user-organisations-listing/UserOrganisationsListing.jsx:52]: Done

File: packages/main/src/pages/protected/create-workspace/LaunchWorkspace.jsx (hunk 13 above)
Comment (MrTaaJ) [reply packages/main/src/pages/protected/create-workspace/LaunchWorkspace.jsx:33]: Done

File: packages/main/src/pages/protected/create-workspace/LaunchWorkspace.jsx (hunk 14 above)
Comment (MrTaaJ) [reply packages/main/src/pages/protected/create-workspace/LaunchWorkspace.jsx:85]: Done

File: packages/main/src/pages/protected/create-workspace/LaunchWorkspace.jsx (hunk 15 above)
Comment (MrTaaJ) [reply packages/main/src/pages/protected/create-workspace/LaunchWorkspace.jsx:87]: Done

File: packages/main/src/pages/protected/create-workspace/Step0.jsx (hunk 16 above)
Comment (MrTaaJ) [reply packages/main/src/pages/protected/create-workspace/Step0.jsx:25]: Done

File: packages/main/src/pages/protected/workspace/useParamHook.jsx (hunk 17 above)
Comment (MrTaaJ) [reply packages/main/src/pages/protected/workspace/useParamHook.jsx:35]: Done

**Comment by MrTaaJ** [comment]
All the new changes have been implemented. For the null, I saw it used in the code base and I think it is a good way of saving the currentWorkspace to avoid it being undefined. I have made all the necessary changes and removed the null as advised.

File: packages/main/src/components/protected/topbar/components/Profile.jsx [hunk 18]
```diff
@@ -46,7 +46,7 @@ const Profile = () => {
   useEffect(() => {
     if (currentWorkspace) {
       authAxios
-        .get(`/organizations/${currentWorkspace}`)
+        .get(`/organization/${currentWorkspace}`)
```
Comment (JulianaSau) [inline packages/main/src/components/protected/topbar/components/Profile.jsx:49]: This should be `/organisations/${currentWorkspace}`

**Review by JulianaSau — CHANGES_REQUESTED** [review]

File: packages/root-config/src/zuri-root-config.js (hunk 6 above)
Comment (MrTaaJ) [reply packages/root-config/src/zuri-root-config.js:41]: Done

Comment (MrTaaJ) [reply packages/root-config/src/zuri-root-config.js:41]: Done

File: packages/main/src/components/protected/topbar/components/Profile.jsx (hunk 18 above)
Comment (MrTaaJ) [reply packages/main/src/components/protected/topbar/components/Profile.jsx:49]: Seen

**Comment by MrTaaJ** [comment]
Done

**Review by JulianaSau — APPROVED** [review]

**Pull request merged**